### ISSUE TITLE
`escapeinside` is fragile with some lexers

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#825]
(Original issue 1118 created by g_poore on 2015-06-23T22:26:13.182331+00:00)

If I understand correctly, `escapeinside` works by tokenizing to obtain strings and comments, then merging all non-string and non-comments tokens, then scanning these merged tokens for escapes, and finally scanning everything that wasn't escaped for normal tokens. So it isn't possible to have escapes inside of strings and comments. This triple-scan approach can cause `escapeinside` to fail in non-obvious ways.

For example, in the `bash` lexer the backslash `\` and immediately following character is treated as a `String.Escape` token. Because of this, standard LaTeX commands of the form `\command` will always be treated as starting with a type of string, even when they occur in a region "protected" by `escapeinside`. More details about this particular case are [here](https://github.com/gpoore/minted/issues/70#issuecomment-111729930).

Basically, any time that the LaTeX code contained within an escape looks like a string or comment for the current lexer, `escapeinside` may fail. If it would be possible to tokenize everything in a single stage, with escapes having the highest priority, that might be a solution.

**Comment by cykerway** [comment]
Split escapes by strings is certainly wrong. But the other side is also nontrivial. An escape could be in the middle of another token. Split by escapes cuts the token in halves, which may make the lexer unable to parse each. The correct way of doing things would be the lexer reads the entire token and knows there is an escape in this token. The question of handling this enveloping token is better left to other parts such as the formatter.

**Comment by linzhongzi** [comment]
`'''my test: 提取19 한국어" \''' \\\''' \\\\' \\\\\'' \\\\\\\\'''`
python code in test.py,
pygments  fetch the result  that is  (Token.Literal.String.Doc, '\'\'\'自定义的测试用例: 提取19 한국어" \\\'\'\'')
But actually all the whole setence is Token.Literal.String.Doc,
